### PR TITLE
[D2IQ-71746] Add upgrade tests for all addon test groups

### PR DESCRIPTION
**Pull request by shaneutt**
**What type of PR is this?**
Feature

**What this PR does/ why we need it**:
This adds a default test to CI which will ensure that updated addons are able to upgrade from the previous revision in `master`.

**Which issue(s) this PR fixes**:
https://jira.d2iq.com/browse/D2IQ-71746

**Special notes for your reviewer**:
Blocked on https://github.com/mesosphere/kubeaddons/pull/979

**Does this PR introduce a user-facing change?**:
NONE

**Checklist**

* [x] The commit message explains the changes and why are needed.
* [x] The code builds and passes lint/style checks locally.
* [x] The relevant subset of integration tests pass locally.
* [x] The core changes are covered by tests.
* [x] The documentation is updated where needed.

### CODE REVIEW

File: test/addons_test.go [hunk 1]
```diff
@@ -265,13 +267,54 @@ func testgroup(t *testing.T, groupname string, version string, jobs ...clusterTe
...
+		oldVersion, err := semver.Parse(strings.TrimPrefix(oldAddon.GetAnnotations()[constants.AddonRevisionAnnotation], "v"))
+		if err != nil {
+			return err
+		}
+		newVersion, err := semver.Parse(strings.TrimPrefix(newAddon.GetAnnotations()[constants.AddonRevisionAnnotation], "v"))
+		if err != nil {
+			return err
+		}
+		if oldVersion.GT(newVersion) {
+			return fmt.Errorf("revisions for addon %s are broken, previous revision %s is newer than current %s", newAddon.GetName(), oldVersion, newVersion)
```
Comment (tillt) [inline test/addons_test.go:293]:
```suggestion
			return fmt.Errorf("revisions for addon %s are invalid, previous revision %s is newer than current %s", newAddon.GetName(), oldVersion, newVersion)
```

File: test/addons_test.go [hunk 2]
```diff
@@ -265,13 +267,54 @@ func testgroup(t *testing.T, groupname string, version string, jobs ...clusterTe
...
+		}
+		if !newVersion.GT(oldVersion) {
+			t.Logf("skipping upgrade test for addon %s, it has not be updated", newAddon.GetName())
+			continue
+		}
+
+		t.Logf("INFO: addon %s was modified and will be upgrade tested", newAddon.GetName())
```
Comment (tillt) [inline test/addons_test.go:300]: Same question https://github.com/mesosphere/kubeaddons/pull/979/files#r489378817 :D

**Review by tillt — APPROVED** [review]
lgtm

**Review by armandgrillet — APPROVED** [review]

File: test/addons_test.go (hunk 2 above)
Comment (shaneutt) [reply test/addons_test.go:300]: I followed up over there, let me know if that's satisfactory, of if you'd like to see something change!

**Pull request merged**